vpr/src/draw/draw.cpp

```diff
@@ -1194,27 +1195,27 @@ static void draw_routing_costs(ezgl::renderer* g) {
     float min_cost = std::numeric_limits<float>::infinity();
     float max_cost = -min_cost;
     std::vector<float> rr_node_costs(device_ctx.rr_nodes.size(), 0.);
+
+    /* For display purpose only, not an accurate value of the present congestion factor */
+    float approx_pres_fac = 1.;
```

Comment:
Could you get the accurate value of pres_fac?  Using 1 here could get confusing.  You could put a copy of pres_fac in the draw_state if necessary and update it when or before calling update_screen; not sure if there's a better option.

If you use an approximate value, a large value (say 10) would be better than 1 to ensure the present congestion significantly impacts the display.  But I really think it would be best to get the real value.

Reply:
a subsystem scope variable **pres_fac** added in draw_global.h and draw_global.cpp

Reply:
Actually nvm. I'll put them in draw states like you said